### ISSUE TITLE
mime is private

### ISSUE THREAD

**Issue by ghost** [novacrazy/serde_shims#1]
I include mime using "use mime_serde_shim as mime;"
And when I try to use e.g. "mime::Mime::from_str" I get "error[E0603]: struct `Mime` is private"

**Comment by novacrazy** [comment]
Ah, these aren't substitutions for their original crates, but rather it provides `serialize`/`deserialize` functions that can be used with `#[serde(with = "mime_serde_shim")]` as shown in the example here: https://novacrazy.github.io/serde_shims/mime_serde_shim/index.html

**Comment by ghost** [comment]
Do I need to use a wrapper type?

**Comment by novacrazy** [comment]
No, you just need to point Serde at the shim crate or shim module (if using `serde_shims` meta crate), like so:

```rust
#[derive(Debug, PartialEq, Serialize, Deserialize)]
struct MimeTest {
    #[serde(with = "mime_serde_shim")]
    mime: mime::Mime, // using the original `mime` crate and struct
}
```

or with the whole meta crate (assuming the `mime` feature is enabled):

```rust
#[derive(Debug, PartialEq, Serialize, Deserialize)]
struct MimeTest {
    #[serde(with = "serde_shims::mime")]
    mime: mime::Mime,
}
```

`#[serde(with = "path")]` automatically uses the `serialize/deserialize` functions at those paths.

**Comment by ghost** [comment]
I have a field of type `Option<Vec<mime::Mime>>` and thus that doesn't seem to work on it

**Comment by novacrazy** [comment]
Hmm. That is indeed one of the limitations of doing it this way. Would adding a simple wrapper type that `Deref`'s to `Mime` be a good solution for you?

**Comment by ghost** [comment]
A bit redundancy but should be fine for now.

**Comment by novacrazy** [comment]
I have published version `0.2.2` of `mime_serde_shim` which includes a `Wrapper` type.

You can view the new type and an example here: https://docs.rs/mime_serde_shim/0.2.2/mime_serde_shim/
